transforms/no-vars.js

```diff
@@ -49,24 +86,40 @@ module.exports = function(file, api) {
     .find(j.VariableDeclaration)
     .filter(
       p => {
-        if (p.value.kind !== 'var') return false;
-
         if (
           'ForStatement' === p.parent.value.type ||
           'ForInStatement' === p.parent.value.type ||
-          'ForOfStatement' === p.parent.value.type ||
-          isMutated(p)
+          'ForOfStatement' === p.parent.value.type
         ) {
-          p.value.kind = 'let';
+          if (!isAccessedInClosure(p.parent)) {
+            p.value.kind = 'let';
+          }
```

Comment:
do we want to log and warn the user about the potential race condition in this case?